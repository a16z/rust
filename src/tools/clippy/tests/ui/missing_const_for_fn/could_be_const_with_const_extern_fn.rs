#![warn(clippy::missing_const_for_fn)]
#![allow(unsupported_calling_conventions)]
#![feature(const_extern_fn)]

extern "C-unwind" fn c_unwind() {}
//~^ ERROR: this could be a `const fn`
extern "system" fn system() {}
//~^ ERROR: this could be a `const fn`
extern "system-unwind" fn system_unwind() {}
//~^ ERROR: this could be a `const fn`
pub extern "stdcall" fn std_call() {}
//~^ ERROR: this could be a `const fn`
pub extern "stdcall-unwind" fn std_call_unwind() {}
//~^ ERROR: this could be a `const fn`
