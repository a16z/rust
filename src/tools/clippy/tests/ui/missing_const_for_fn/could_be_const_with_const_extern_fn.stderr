error: this could be a `const fn`
  --> tests/ui/missing_const_for_fn/could_be_const_with_const_extern_fn.rs:5:1
   |
LL | extern "C-unwind" fn c_unwind() {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::missing-const-for-fn` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::missing_const_for_fn)]`
help: make the function `const`
   |
LL | const extern "C-unwind" fn c_unwind() {}
   | +++++

error: this could be a `const fn`
  --> tests/ui/missing_const_for_fn/could_be_const_with_const_extern_fn.rs:7:1
   |
LL | extern "system" fn system() {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: make the function `const`
   |
LL | const extern "system" fn system() {}
   | +++++

error: this could be a `const fn`
  --> tests/ui/missing_const_for_fn/could_be_const_with_const_extern_fn.rs:9:1
   |
LL | extern "system-unwind" fn system_unwind() {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: make the function `const`
   |
LL | const extern "system-unwind" fn system_unwind() {}
   | +++++

error: this could be a `const fn`
  --> tests/ui/missing_const_for_fn/could_be_const_with_const_extern_fn.rs:11:1
   |
LL | pub extern "stdcall" fn std_call() {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: make the function `const`
   |
LL | pub const extern "stdcall" fn std_call() {}
   |     +++++

error: this could be a `const fn`
  --> tests/ui/missing_const_for_fn/could_be_const_with_const_extern_fn.rs:13:1
   |
LL | pub extern "stdcall-unwind" fn std_call_unwind() {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: make the function `const`
   |
LL | pub const extern "stdcall-unwind" fn std_call_unwind() {}
   |     +++++

error: aborting due to 5 previous errors

